File: Desktop/glamai-fyp/backend/app/models/payment.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

from app.database import Base


class Payment(Base):
    __tablename__ = "payments"

    id = Column(Integer, primary_key=True, index=True)
    appointment_id = Column(Integer, ForeignKey("appointments.id"), nullable=False, unique=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    amount = Column(Float, nullable=False)
    payment_method = Column(String, nullable=False)   # esewa | khalti | cash
    payment_status = Column(String, nullable=False, default="completed")
    transaction_ref = Column(String, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    appointment = relationship("Appointment")
    user = relationship("User")

File: Desktop/glamai-fyp/mobile_app/lib/models/payment.dart
class Payment {
  final int id;
  final int appointmentId;
  final int userId;
  final double amount;
  final String paymentMethod;   // esewa | khalti | cash
  final String paymentStatus;   // completed
  final String transactionRef;
  final String? createdAt;

  const Payment({
    required this.id,
    required this.appointmentId,
    required this.userId,
    required this.amount,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.transactionRef,
    this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id'] as int,
        appointmentId: json['appointment_id'] as int,
        userId: json['user_id'] as int,
        amount: (json['amount'] as num).toDouble(),
        paymentMethod: json['payment_method'] as String,
        paymentStatus: json['payment_status'] as String,
        transactionRef: json['transaction_ref'] as String,
        createdAt: json['created_at'] as String?,
      );

  String get methodLabel => switch (paymentMethod.toLowerCase()) {
        'esewa' => 'eSewa',
        'khalti' => 'Khalti',
        _ => 'Cash on Arrival',
      };
}
